### PR TITLE
HPCC-18222 DESDL refresh issue when overwriting

### PR DESCRIPTION
**Pull request by miguelvazq**
Whenever a navigates to a binding performs an action and then leaves and tries to perform another action the data from previous selection is still active.

Signed-off by: Miguel Vazquez <miguel.vazquez@lexisnexis.com>

## Type of change:
- [x] This change is a bug fix (non-breaking change which fixes an issue).
- [ ] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change improves the code (refactor or other change that does not change the functionality)
- [ ] This change fixes warnings (the fix does not alter the functionality or the generated code)
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).
- [ ] This change alters the query API (existing queries will have to be recompiled)

## Checklist:
- [x] My code follows the code style of this project.
  - [ ] My code does not create any new warnings from compiler, build system, or lint.
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit is signed.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the CONTRIBUTORS document.
- [ ] The change has been fully tested:
  - [ ] I have added tests to cover my changes.
  - [ ] All new and existing tests passed.
  - [ ] I have checked that this change does not introduce memory leaks.
  - [ ] I have used Valgrind or similar tools to check for potential issues.
- [ ] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] Existing deployed queries will not be broken
  - [ ] This change fixes the problem, not just the symptom
  - [ ] The target branch of this pull request is appropriate for such a change.
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised JIRA issues to address them separately
- [x] This is a user interface / front-end modification
  - [x] I have tested my changes in multiple modern browsers
  - [x] The component(s) render as expected

### CODE REVIEW

File: esp/src/eclwatch/DynamicESDLMethodWidget.js
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLMethodWidget.js]: The Delay loader will spot and call "refresh" with the new params as needed, so do not remove the call to the base class, instead change:
```
         refresh: function () {
             this.refreshGrid();
         },
```
to:
```
         refresh: function (params) {
             this._params = params;
             this.refreshGrid();
         },
```
(You may need to also call the refreshState in there as well)

**Review by GordonSmith — CHANGES_REQUESTED** [review]

**Comment by miguelvazq** [comment]
@GordonSmith Modified.

File: esp/src/eclwatch/DynamicESDLMethodWidget.js
Comment (GordonSmith) [inline esp/src/eclwatch/DynamicESDLMethodWidget.js]: You may want to move the call to ```this._refreshActionState();``` to inside the refresh function?

Comment (miguelvazq) [reply esp/src/eclwatch/DynamicESDLMethodWidget.js]: Changed. Is this call required even if no event calls are being made on this particular grid?

**Comment by miguelvazq** [comment]
@richardkchapman Tested and performing as expected. Had a minor question, OK to merge once approved.

**Review by GordonSmith — APPROVED** [review]
Looks good.

**Comment by miguelvazq** [comment]
@richardkchapman Ready to merge.

**Pull request merged**
